ic-os/components/setupos-scripts/test-setupos.sh

```diff
@@ -0,0 +1,251 @@
+# Unit tests for check-network.sh
+# ------------------------------------------------------------------------------
+
+function test_validate_domain_name() {
+    echo "Running test: test_validate_domain_name_valid"
+    domain_name="example.com"
+    if validate_domain_name; then
+        echo "  PASS: valid domain: $domain_name"
+    else
+        echo "  FAIL: valid domain: $domain_name"
+        exit 1
+    fi
+
+    echo "Running test: test_validate_domain_name_invalid"
+    domain_name="example."
+    if ! (validate_domain_name); then
+        echo "  PASS: domain $domain_name validation failed as expected"
+    else
+        echo "  FAIL: domain $domain_name validation was expected to fail but didn't"
+        exit 1
+    fi
+
+    domain_name="&BadDOMAIN.com"
+    if ! (validate_domain_name); then
+        echo "  PASS: domain $domain_name validation failed as expected"
+    else
+        echo "  FAIL: domain $domain_name validation was expected to fail but didn't"
+        exit 1
+    fi
+}
+
+# ------------------------------------------------------------------------------
+# Unit tests for check-hardware.sh
+# ------------------------------------------------------------------------------
+
+function test_detect_hardware_generation_gen1() {
+    echo "Running test: test_detect_hardware_generation_gen1"
+    FAKE_CPU_JSON='[
+      {"id": "cpu:0", "product": "AMD EPYC 7302", "capabilities": {"sev": "true"}},
+      {"id": "cpu:1", "product": "AMD EPYC 7302", "capabilities": {"sev": "true"}}
+    ]'
+    function get_cpu_info_json() { echo "$FAKE_CPU_JSON"; }
+    HARDWARE_GENERATION=""
+
+    detect_hardware_generation
+    if [[ "$HARDWARE_GENERATION" == "1" ]]; then
+        echo "  PASS: Gen1 hardware detected"
+    else
+        echo "  FAIL: Gen1 hardware not detected as expected"
+        exit 1
+    fi
+}
+
+function test_detect_hardware_generation_gen2() {
+    echo "Running test: test_detect_hardware_generation_gen2"
+    FAKE_CPU_JSON='[
+      {"id": "cpu:0", "product": "AMD EPYC 7313", "capabilities": {"sev_snp": "true"}},
+      {"id": "cpu:1", "product": "AMD EPYC 7313", "capabilities": {"sev_snp": "true"}}
+    ]'
+    function get_cpu_info_json() { echo "$FAKE_CPU_JSON"; }
+    HARDWARE_GENERATION=""
+
+    detect_hardware_generation
+    if [[ "$HARDWARE_GENERATION" == "2" ]]; then
+        echo "  PASS: Gen2 hardware detected"
+    else
+        echo "  FAIL: Gen2 hardware not detected as expected"
+        exit 1
+    fi
+}
+
+function test_verify_cpu_gen1() {
+    echo "Running test: test_verify_cpu_gen1"
+    FAKE_CPU_JSON='[
+      {"id": "cpu:0", "product": "AMD EPYC 7302", "capabilities": {"sev": "true"}},
+      {"id": "cpu:1", "product": "AMD EPYC 7302", "capabilities": {"sev": "true"}}
+    ]'
+    function get_cpu_info_json() { echo "$FAKE_CPU_JSON"; }
+    function nproc() { echo 64; }
+    HARDWARE_GENERATION="1"
+
+    verify_cpu
+    echo "  PASS: verify_cpu for Gen1 passed"
```

Comment:
Why don't you catch the error here as well and print an explicit "  FAIL: verify_cpu for Gen1 failed unexpectedly" such that it is consistent across the tests?